File: 1013-fibonacci-number/fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n<2){
            return n;
        }
        int s1=0,s2=1;
        for(int i=2;i<=n;i++){
            int next=s1+s2;
            s1=s2;
            s2=next;
        }
        return s2;
    }
}

File: 1013-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        dict={}
        def fibo(n):
            if n<2:
                return n
            if(n in dict.keys()):
                return dict[n]
            dict[n]= fibo(n-1)+fibo(n-2)
            return dict[n]
        return fibo(n)
        